Library/Homebrew/extend/os/mac/utils/bottles.rb

```diff
@@ -6,9 +6,11 @@ class << self
       undef tag
 
       def tag
-        tag = MacOS.version.to_sym
-        tag = "#{tag}_arm".to_sym if Hardware::CPU.arm?
-        tag
+        if Hardware::CPU.intel?
+          MacOS.version.to_sym
+        else
+          "#{ENV["HOMEBREW_PROCESSOR"]}_#{MacOS.version}".to_sym
```

Comment:
Just to keep up with the most recent conversation: what does this output on 
Apple Silicon, currently? What does it output on Linux-on-ARM?

Reply:
Apple Silicon (on `master`):

```
irb(main):001:0> Utils::Bottles.tag
=> :big_sur_arm
irb(main):002:0> "#{ENV["HOMEBREW_PROCESSOR"]}_#{MacOS.version}".to_sym
=> :"arm_11.0"
```

Apple Silicon (on this PR branch):

```
irb(main):001:0> Utils::Bottles.tag
=> :"arm64_11.0"
irb(main):002:0> "#{ENV["HOMEBREW_PROCESSOR"]}_#{MacOS.version}".to_sym
=> :"arm64_11.0"
```

Reply:
Thanks! In that case:

```suggestion
          "#{ENV["HOMEBREW_PROCESSOR"]}_#{MacOS.version.to_sym}".to_sym
```

At least so we're not referring to it as `11.0` (which isn't even a "valid" symbol).

Does Apple refer to it as ARM64 much/at all?

Reply:
> Does Apple refer to it as ARM64 much/at all?

They’re going to great lengths to avoid any mention of ARM in their public-facing copy.

Reply:
Given that I wonder about using one of the terms they actually use? Can you summarise them at all @claui?

Reply:
Apple Silicon is the brand they use but when they talk about the architecture they use `arm64` exclusively. No `aarch64` even though it's the same thing.

e.g. https://developer.apple.com/documentation/xcode/porting_your_macos_apps_to_apple_silicon

Reply:
Gotcha. I'd be fine with `arm64_big_sur` or `apple_silicon_big_sur`.

My reasoning for `big_sur_arm` was that we will likely end up having a 11.x where they/we don't support x86_64 any longer.

Reply:
I prefer `arm64` over `apple_silicon`, because it's what's returned by `uname -m`, and it's many characters shorter.

Reply:
Ready for re-review.